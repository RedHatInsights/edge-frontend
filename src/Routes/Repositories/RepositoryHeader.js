import React from 'react';
import PageHeader, {
  PageHeaderTitle,
} from '@redhat-cloud-services/frontend-components/PageHeader';
import { Popover } from '@patternfly/react-core';
import QuestionCircleIcon from '@patternfly/react-icons/dist/js/icons/outlined-question-circle-icon';
//import { ExternalLinkAltIcon } from "@patternfly/react-icons";

const RepositoryHeader = () => {
  return (
    <PageHeader>
      <>
        <PageHeaderTitle title="Applications settings" />
        <p className="pf-u-mt-sm">
          Settings for Edge Management
          <Popover
            aria-label="Basic popover"
            headerContent={<div>About Edge Management</div>}
            bodyContent={
              <div>
                Edge Management is a service that allows you to provision,
                update and maintain edge systems.
              </div>
            }
            //footerContent={
            //  <Button variant="link" style={{ 'padding-left': 0 }}>
            //    <a href="#">Documentation</a>
            //    <ExternalLinkAltIcon className="pf-u-ml-sm" />
            //  </Button>
            //}
          >
            <QuestionCircleIcon
              style={{ cursor: 'pointer' }}
              className="pf-u-ml-xs"
              data-testid="test-pop-over"
            />
          </Popover>
        </p>
      </>
    </PageHeader>
  );
};

export default RepositoryHeader;
