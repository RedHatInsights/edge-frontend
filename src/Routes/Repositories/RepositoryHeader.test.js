import React from 'react';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import RepositoryHeader from './RepositoryHeader';

describe('RepositoryHeader', () => {
  it('renders correctly', async () => {
    const { container } = render(<RepositoryHeader />);

    expect(screen.getByRole('heading', 'Applications settings')).toBeDefined();
    expect(screen.getByText('Settings for Edge Management')).toBeDefined();
    expect(() => screen.getByText('About Edge Management')).toThrow();
    expect(() =>
      screen.getByText(
        'Edge Management is a service that allows you to provision, update and maintain edge systems.'
      )
    ).toThrow();
    fireEvent.click(screen.getByTestId('test-pop-over'));
    await waitFor(() =>
      expect(screen.getByText('About Edge Management')).toBeDefined()
    );
    await waitFor(() =>
      expect(
        screen.getByText(
          'Edge Management is a service that allows you to provision, update and maintain edge systems.'
        )
      ).toBeDefined()
    );

    expect(container.querySelector('div')).toMatchSnapshot();
  });
});
