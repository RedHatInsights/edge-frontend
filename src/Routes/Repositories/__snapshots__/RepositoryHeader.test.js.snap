// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepositoryHeader renders correctly 1`] = `
<div
  class="pf-v5-l-flex pf-m-justify-content-space-between"
>
  <div
    class=""
  >
    <h1
      class="pf-v5-c-title pf-m-2xl"
      data-ouia-component-id="OUIA-Generated-RHI/Header-true-2"
      data-ouia-component-type="RHI/Header"
      data-ouia-safe="true"
      widget-type="InsightsPageHeaderTitle"
    >
      Custom repositories
    </h1>
  </div>
</div>
`;
